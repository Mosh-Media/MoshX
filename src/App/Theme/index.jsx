import React, { useMemo } from "react";

import { ThemeProvider } from "@material-ui/core/styles";
import CssBaseline from '@material-ui/core/CssBaseline';
import getTheme from "../../theme";
import useGlobal from "store";

const Theme = ({ children }) => {
    const [isDarkMode] = useGlobal(state => state.themeOptions.isDarkMode);

    const theme = useMemo(() => getTheme({ isDarkMode }), [isDarkMode]);

    return (
        <ThemeProvider theme={theme}>
            <CssBaseline />
            {children}
        </ThemeProvider>
    )
}

export default React.memo(Theme)