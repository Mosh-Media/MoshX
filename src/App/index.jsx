/**
 *
 *
 * @author Mo Skool
 * @version 1.0.0
 * @visibleName App Component ðŸ¥ˆ
 *
 *
 * App component handles the following:
 *
 *  - lazy loads pages/containers
 *  - Handles routing for app
 *  - Provides the app with Material-UI theme instance
 *  - Renders App navigation
 *  - Suspense and loading
 */
import React, { Suspense, lazy } from "react";

import * as ROUTES from "constants/routes";
import { BrowserRouter as Router, Route, Switch } from "react-router-dom";
import MoSpinner from "components/library/MoSpinner";
import withStoreInit from "components/shared/Session/withStoreInit";

const Account = lazy(() => import("containers/Account"));
const AdminPage = lazy(() => import("containers/AdminPage"));
const LandingPage = lazy(() => import("containers/Landing"));
const NotFound = lazy(() => import("components/shared/NotFound"));
const PasswordReset = lazy(() => import("containers/PasswordReset"));

const Playground = lazy(() => import("containers/Playground"));
const Collections = lazy(() => import("containers/Collections"));
const SignUp = lazy(() => import("containers/SignUp"));
const SignIn = lazy(() => import("containers/SignIn"));

const App = () => {
  return (
    <Router>
      {/* Page size based on the page */}
      <Suspense fallback={<MoSpinner isLoading={true} color="primary" />}>
        <Switch>
          <Route path={ROUTES.ADMIN.path} component={AdminPage} />
          <Route path={ROUTES.ACCOUNT.path} component={Account} />
          <Route exact path={ROUTES.LANDING.path} component={LandingPage} />
          <Route exact path={ROUTES.NOT_FOUND.path} component={NotFound} />

          <Route path={ROUTES.SIGN_IN.path} component={SignIn} />
          <Route path={ROUTES.SIGN_UP.path} component={SignUp} />
          <Route path={ROUTES.PASSWORD_FORGET.path} component={PasswordReset} />
          <Route path={ROUTES.PLAYGROUND.path} component={Playground} />
          <Route path={ROUTES.COLLECTIONS.path} component={Collections} />
          <Route path={ROUTES.COMPANIES.path} component={Collections} />
          <Route component={NotFound} />
        </Switch>
      </Suspense>
    </Router>
  );
};

export default withStoreInit(App);
