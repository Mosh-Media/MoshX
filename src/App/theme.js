import createMuiTheme from "@material-ui/core/styles/createMuiTheme";
import responsiveFontSizes from "@material-ui/core/styles/responsiveFontSizes";

const bigShadow =
	"0 24px 24px -18px rgba(69,104,129,.33), 0 9px 45px 0 rgba(114,119,160,.12)";

const colors = {
	grey: {
		light: "#a6a6a6",
		medium: "#424B54",
		dark: "#383c40",
		darkest: "#2b2b2b"
	},
	green: {
		light: "#41D3BD",
		dark: "#157A6E",
		darkest: ""
	},
	yellow: {
		light: "#F5D547",
		dark: "",
		darkest: ""
	},
	white: {
		medium: "#FFFFF2"
	},
	blue: {
		medium: "#276ef1"
	}
};

const padding = {
  xs: "4px",
  sm: "12px",
  md: "36px",
  lg: "48px",
  xl: "96px"
}

let theme = createMuiTheme({
	bigShadow: bigShadow,
	card: {
		boxShadow: bigShadow,
		"&:hover": {
			transform: "translateY(-5px)",
			backgroundColor: "#FFF"
		},
		"&:hover .desc": {
			color: "white"
		},
		transition:
			"transform .35s cubic-bezier(.4,0,.2,1),box-shadow .35s cubic-bezier(.4,0,.2,1)",
		transform: "translateY(0)"
	},
	container: {
		width: "calc(100vw - 28.799999999999997rem)"
	},
	flexAlignCenter: {
		display: "flex",
		alignItems: "center"
	},
	padding: padding,
	grey: colors.grey,
	green: colors.green,
	yellow: colors.yellow,
	white: colors.white,
	blue: colors.blue
});

const fontOptions = {
	breakpoints: ["sm", "md", "lg", "xl"]
};

theme = responsiveFontSizes(theme, fontOptions);

export default theme;
