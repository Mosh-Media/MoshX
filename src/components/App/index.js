import React from "react";
import { BrowserRouter as Router, Route } from "react-router-dom";

import Navigation from "../Navigation";
import LandingPage from "../Landing";
import SignUp from "../SignUp";
import SignIn from "../SignIn";
import PasswordForgot from "../PasswordForgot";
import Home from "../Home";
import Account from "../Account";
import Admin from "../Admin";

import * as ROUTES from "../../constants/routes";
import { withAuthentication } from "../Session";
import Container from "@material-ui/core/Container";

const App = () => (
	<Router>
		<Navigation />
		<Container maxWidth="xl" className="container">
			<Route exact path={ROUTES.LANDING} component={LandingPage} />
			<Route path={ROUTES.SIGN_UP} component={SignUp} />
			<Route path={ROUTES.SIGN_IN} component={SignIn} />
			<Route path={ROUTES.PASSWORD_FORGET} component={PasswordForgot} />
			<Route path={ROUTES.HOME} component={Home} />
			<Route path={ROUTES.ACCOUNT} component={Account} />
			<Route path={ROUTES.ADMIN} component={Admin} />
			<Route path={ROUTES.NEW_PROJECT} component={Admin} />
			{/* <div>Icons made by <a href="https://www.flaticon.com/authors/freepik" title="Freepik">Freepik</a> from <a href="https://www.flaticon.com/" title="Flaticon">www.flaticon.com</a></div> */}
		</Container>
	</Router>
);

export default withAuthentication(App);
