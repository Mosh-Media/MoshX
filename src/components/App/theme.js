import { createMuiTheme } from "@material-ui/core/styles";

const bigShadow =
	"0 24px 24px -18px rgba(69,104,129,.33), 0 9px 45px 0 rgba(114,119,160,.12)";

const colors = {
	grey: "#a6a6a6",
	green: "#41D3BD",
};
const theme = createMuiTheme({
	bigShadow: bigShadow,
	card: {
		boxShadow: bigShadow,
		"&:hover": {
			transform: "translateY(-5px)",
			backgroundColor: "#FFF"
		},
		"&:hover .desc": {
			color: "white"
		},
		transition:
			"transform .35s cubic-bezier(.4,0,.2,1),box-shadow .35s cubic-bezier(.4,0,.2,1)",
		transform: "translateY(0)"
	},
	grey: colors.grey,
	green: colors.green,
});

export default theme;
