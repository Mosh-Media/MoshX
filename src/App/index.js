import React, { Suspense, lazy } from "react";

import * as ROUTES from "constants/routes";
import * as Sentry from "@sentry/browser";

import { BrowserRouter as Router, Route, Switch } from "react-router-dom";
import MoSpinner from "components/shared/MoSpinner";
import theme from "./theme";
import { ThemeProvider } from "@material-ui/core/styles";
import { withAuthentication } from "components/Session";

const Navigation = lazy(() => import("components/Navigation"));
const Account = lazy(() => import("containers/Account"));
const AdminPage = lazy(() => import("containers/AdminPage"));
const Container = lazy(() => import("@material-ui/core/Container"));
const Home = lazy(() => import("containers/Home"));
const LandingPage = lazy(() => import("containers/Landing"));
const NotFound = lazy(() => import("components/NotFound"));
const PasswordForgot = lazy(() => import("components/PasswordForgot"));
const Playground = lazy(() => import("containers/Playground"));
const Course = lazy(() => import("containers/Course"));
const Courses = lazy(() => import("containers/Courses"));
const QuestionViewMode = lazy(() =>
  import("containers/Question/QuestionViewMode")
);
const QuestionEditMode = lazy(() =>
  import("containers/Question/QuestionEditMode")
);
const SignUp = lazy(() => import("containers/SignUp"));
const SignIn = lazy(() => import("containers/SignIn"));

const App = (props) => (
  <ThemeProvider theme={theme}>
    <Router>
      <Suspense fallback={<MoSpinner isLoading={true} color="primary" />}>
        <Container style={{ height: "100%", maxHeight: "100%" }}>
          <Navigation {...props} />
          <Switch>
            <Route path={ROUTES.ADMIN.path} component={AdminPage} />
            <Route path={ROUTES.ACCOUNT.path} component={Account} />
            <Route path={ROUTES.HOME.path} component={Home} />
            <Route exact path={ROUTES.LANDING.path} component={LandingPage} />
            <Route exact path={ROUTES.NOT_FOUND.path} component={NotFound} />

            <Route path={ROUTES.SIGN_IN.path} component={SignIn} />
            <Route path={ROUTES.SIGN_UP.path} component={SignUp} />
            <Route
              path={ROUTES.PASSWORD_FORGET.path}
              component={PasswordForgot}
            />
            <Route path={ROUTES.PLAYGROUND.path} component={Playground} />
            <Route exact path={ROUTES.COLLECTIONS.path} component={Courses} />
            <Route
              exact
              path={ROUTES.COLLECTIONS.path + "/:collection"}
              component={Course}
            />
            <Route
              exact
              path={ROUTES.COLLECTIONS.path + "/:collection/:questionId"}
              component={QuestionViewMode}
            />
            <Route
              exact
              path={
                ROUTES.COLLECTIONS.path + "/:collection/:questionId/isEditMode"
              }
              component={QuestionEditMode}
            />
            <Route component={NotFound} />
          </Switch>
        </Container>
      </Suspense>
    </Router>
  </ThemeProvider>
);

if (process.env.NODE_ENV === "production") {
  Sentry.init({
    dsn: "https://2cb4b0fa634941a69b5bdd868a07a024@sentry.io/1878459",
  });
}

export default withAuthentication(App);
