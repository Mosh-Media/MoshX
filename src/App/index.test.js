import React from "react";
import renderer from "react-test-renderer";
import App from "./index";

it("Should render App", () => {
  const tree = renderer.create(<App />).toJSON();
  expect(tree).toMatchSnapshot();
});
