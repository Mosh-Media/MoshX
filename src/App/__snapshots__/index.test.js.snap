// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component should match the snapshot 1`] = `
<BrowserRouter>
  <React.Suspense
    fallback={
      <ForwardRef(WithStyles)
        color="primary"
        isLoading={true}
      />
    }
  >
    <UNDEFINED />
    <UNDEFINED
      maxWidth="lg"
    >
      <Switch>
        <Route
          component={
            Object {
              "$$typeof": Symbol(react.lazy),
              "_ctor": [Function],
              "_result": null,
              "_status": -1,
            }
          }
          path="/admin"
        />
        <Route
          component={
            Object {
              "$$typeof": Symbol(react.lazy),
              "_ctor": [Function],
              "_result": null,
              "_status": -1,
            }
          }
          path="/account"
        />
        <Route
          component={
            Object {
              "$$typeof": Symbol(react.lazy),
              "_ctor": [Function],
              "_result": null,
              "_status": -1,
            }
          }
          exact={true}
          path="/"
        />
        <Route
          component={
            Object {
              "$$typeof": Symbol(react.lazy),
              "_ctor": [Function],
              "_result": null,
              "_status": -1,
            }
          }
          exact={true}
          path="/not_found"
        />
        <Route
          component={
            Object {
              "$$typeof": Symbol(react.lazy),
              "_ctor": [Function],
              "_result": null,
              "_status": -1,
            }
          }
          path="/signin"
        />
        <Route
          component={
            Object {
              "$$typeof": Symbol(react.lazy),
              "_ctor": [Function],
              "_result": null,
              "_status": -1,
            }
          }
          path="/signup"
        />
        <Route
          component={
            Object {
              "$$typeof": Symbol(react.lazy),
              "_ctor": [Function],
              "_result": null,
              "_status": -1,
            }
          }
          path="/pw-forget"
        />
        <Route
          component={
            Object {
              "$$typeof": Symbol(react.lazy),
              "_ctor": [Function],
              "_result": null,
              "_status": -1,
            }
          }
          path="/playgound"
        />
        <Route
          component={
            Object {
              "$$typeof": Symbol(react.lazy),
              "_ctor": [Function],
              "_result": null,
              "_status": -1,
            }
          }
          exact={true}
          path="/courses"
        />
        <Route
          component={
            Object {
              "$$typeof": Symbol(react.lazy),
              "_ctor": [Function],
              "_result": null,
              "_status": -1,
            }
          }
          exact={true}
          path="/courses/:collection"
        />
        <Route
          component={
            Object {
              "$$typeof": Symbol(react.lazy),
              "_ctor": [Function],
              "_result": null,
              "_status": -1,
            }
          }
          path="/courses/:collection/:questionId"
        />
        <Route
          component={
            Object {
              "$$typeof": Symbol(react.lazy),
              "_ctor": [Function],
              "_result": null,
              "_status": -1,
            }
          }
        />
      </Switch>
    </UNDEFINED>
  </React.Suspense>
</BrowserRouter>
`;
